openlibrary/catalog/marc/parse.py

```diff
@@ -212,47 +212,61 @@ def read_work_titles(rec):
             found.append(remove_trailing_dot(' '.join(title).strip(',')))
     if tag_130 := rec.get_fields('130'):
         for f in tag_130:
-            title = ' '.join(
-                v for k, v in f.get_all_subfields() if k.islower() and k != 'n'
+            title = title_from_list(
+                [v for k, v in f.get_all_subfields() if k.islower() and k != 'n']
             )
-            found.append(remove_trailing_dot(title.strip(',')))
+            found.append(title)
     return remove_duplicates(found)
 
 
-def read_title(rec):
+def title_from_list(title_parts, delim=' '):
     # For cataloging punctuation complexities, see https://www.oclc.org/bibformats/en/onlinecataloging.html#punctuation
     STRIP_CHARS = r' /,;:='  # Typical trailing punctuation for 245 subfields in ISBD cataloging standards
+    return delim.join(remove_trailing_dot(s.strip(STRIP_CHARS)) for s in title_parts)
+
+
+def read_title(rec):
```

Comment:
```suggestion
def read_title(rec: BinaryDataField) -> dict[str, str]:
```
Or is `rec` some other datatype?